### PR TITLE
INTMDB-843: [Terraform] Improve acceptance test setup to run in parallel & against cloud-dev. Clusters tests

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

This PR moves the Clusters tests from the old workflow (~ 4 hours to run) to a new one running the tests concurrently and against cloud-dev (~ 41 minutes). 

The new workflow splits the cluster tests into two GitHub Jobs, once for advanced clusters and once for legacy clusters. 

GitHub Action: [5276501597](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/5276501597?pr=1234)

Screenshot 2023-06-15 at 10 12 26

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Further comments
Screenshot 2023-06-14 at 11 47 45

### CODE REVIEW

File: mongodbatlas/data_source_mongodbatlas_advanced_cluster_test.go [hunk 1]
```diff
@@ -1,148 +0,0 @@
...
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
-	"github.com/mwielbut/pointy"
-	matlas "go.mongodb.org/atlas/mongodbatlas"
-)
-
-func TestAccClusterDSAdvancedCluster_basic(t *testing.T) {
-	var (
-		cluster        matlas.AdvancedCluster
-		resourceName   = "mongodbatlas_advanced_cluster.test"
-		dataSourceName = "data.mongodbatlas_advanced_cluster.test"
```
Comment (andreaangiolillo) [inline mongodbatlas/data_source_mongodbatlas_advanced_cluster_test.go:18]: I have moved the data source tests into the resource tests. By doing this, there is no need to create new clusters specifically for these tests. Instead, we can efficiently reuse the clusters that are already created for the resource tests.

File: mongodbatlas/data_source_mongodbatlas_advanced_clusters_test.go [hunk 2]
```diff
@@ -1,151 +0,0 @@
-package mongodbatlas
-
```
Comment (andreaangiolillo) [inline mongodbatlas/data_source_mongodbatlas_advanced_clusters_test.go:2]: I have moved the data source tests into the resource tests. By doing this, there is no need to create new clusters specifically for these tests. Instead, we can efficiently reuse the clusters that are already created for the resource tests.

File: .github/workflows/automated-test-acceptances.yml [hunk 3]
```diff
@@ -85,36 +85,6 @@ jobs:
...
-          MONGODB_ATLAS_TEAMS_IDS: ${{ secrets.MONGODB_ATLAS_TEAMS_IDS }}
-          AWS_REGION: ${{ secrets.AWS_REGION }}
-          CA_CERT: ${{ secrets.CA_CERT }}
-          MONGODB_ATLAS_ENABLE_BETA: ${{ secrets.MONGODB_ATLAS_ENABLE_BETA }}
-          ACCTEST_TIMEOUT: ${{ secrets.ACCTEST_TIMEOUT }}
-          MONGODB_ATLAS_API_KEYS_IDS: ${{ secrets.MONGODB_ATLAS_API_KEYS_IDS }}
-          MONGODB_ATLAS_NETWORK_PROJECT_ID: ${{ secrets.MONGODB_ATLAS_NETWORK_PROJECT_ID }}
-          TEST_REGEX: "^TestAccCluster"
-        run: make testacc
-
```
Comment (andreaangiolillo) [inline .github/workflows/automated-test-acceptances.yml:117]: removing the old tests

**Review by gssbzn — APPROVED** [review]

**Review by Zuhairahmed — APPROVED** [review]

**Review by maastha — APPROVED** [review]
LGTM

**Review by themantissa — APPROVED** [review]
LGTM!  Nice to see the time reduced so much!

**Pull request merged**